src/components/RequestCodePanel.astro

```diff
@@ -0,0 +1,24 @@
+---
+const { title = 'Request', filename } = Astro.props;
+import { Code } from 'astro:components';
+import CodePanel from './CodePanel.astro';
+import createRequestSnippet from '../utils/createRequestSnippet.js';
+
+const file = await import(`../content/api/_code-snippets/${filename}.js`);
+const snippet = createRequestSnippet(file.default);
```

Comment:
@MeganSteenkamp How much more of this could be encapsulated within our module?

```js
// const snippet = centrapaySnippet.convert(filename, target, clientId);

const curlSnippet = centrapaySnippet.convert(filename, 'shell', 'curl');
```

Reply:
@tom-blake I didn't do more because I think we just want the snippet object back.
The module could read the file though.

Having the snippet means we can get the path and the filename which we use in our object:
```
const request = snippet.requests[0];
const path = request.uriObj.pathname;
const method = request.method;
```

Reply:
@MeganSteenkamp Thanks for the context 👍  I think our components shouldn't need to care about low-level httpsnippet details.

I see these new files as the declarative format of our API reference going forward - eventually everything will be in them. `path` and the `method` could be included in the data structure that our converter returns too:

```js
{
    title: 'Invitations',
    description: '...',
    path: '/api/invitations',
    method: 'POST',
    properties: [
        { name: '...', required: true, ..., },
    ],
    request: '<request string here>', // this is the httpsnippet output
    response: '<response string here>'
}
```

Thoughts?

Reply:
@tom-blake I see what you're saying - I think this makes sense. 
I'll update this later today 🫡

Reply:
@tom-blake I've updated this now 🫠